src/domain/application/components/ApplicationContacts.tsx

```diff
@@ -76,32 +77,51 @@ const CustomerFields: React.FC<{
 
   const applicationType = getValues('applicationData.applicationType');
   const selectedContactType = watch(`applicationData.${customerType}.customer.type`);
+  const registryKey = watch(`applicationData.${customerType}.customer.registryKey`);
+  const registryKeyHidden = watch(`applicationData.${customerType}.customer.registryKeyHidden`);
   const registryKeyInputEnabled = isRegistryKeyInputEnabled(
     customerType,
     selectedContactType,
     applicationType,
   );
   const isMounted = useRef(false);
+  const [originalRegistryKeyIsHidden, setOriginalRegistryKeyIsHidden] = useState(false);
 
   useEffect(() => {
-    // If setting contact type disables the registry key or contact type is changed (after mount), clear the registry key
-    if (
-      !isRegistryKeyInputEnabled(customerType, selectedContactType, applicationType) ||
-      isMounted.current
-    ) {
+    // set a flag to mark the original registry key as hidden value in order to be able to revert back to it
+    if (registryKey === HIDDEN_FIELD_VALUE) {
+      setOriginalRegistryKeyIsHidden(true);
```

Comment:
I added this in order to be able to revert changes back to `*******` in case that there is no changes in contact type.